pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+        """
+        Implements a server reactor for
+        :meth:`~pywbem.WBEMConnection.CreateInstance`
+
+        Create a CIM instance in the local repository of this class.
+
+        Exceptions:
+            CIMError CIM_ERR_ALREADY_EXISTS, CIM_ERR_INVALID_CLASS
+        """
+        modified_instance = params['ModifiedInstance']
+        # class_name = modified_instance.classname
+        # TODO class must exist
+
+        inst_name = modified_instance.path
+
+        # classname in instance and path must match
+
+        orig_inst = self._get_instance(inst_name, namespace, **params)
+
+        # ignore IncludeQualifiers as it has been deprecated and its use
+        # by servers is not determinate.
+
+        property_list = params['PropertyList']
+
+        if property_list is not None:
+            self._filter_properties(modified_instance, property_list)
+            # TODO No to following
+            modified_instance.update(modified_instance.path)
+
+        # self._set_instance(cim_class=cimClass
+        #                  instance=modifiedInstance,
+        #                  previous_instance=previousInstance,
+        #                  )
+
+        # find instance in list
+        for inst in self.instances[namespace]:
+            if inst.path == orig_inst.path:
+                inst = modified_instance
+                return
+        raise CIMError(CIM_ERR_FAILED, "Instance could not be updated")
+
+    def _fake_getinstance(self, namespace, **params):
+        """
+        Implements a mock server reactor for
+        :meth:`~pywbem.WBEMConnection.GetInstance`.
+
+        Gets a single instance from the repository based on the
+        InstanceName and filters it for PropertyList, etc.
+
+        This method uses a common method _get_instance to get and
+        process the instance.
+
+        Exceptions:
+            CIMError  CIM_ERR_INVALID_NAMESPACE, CIM_ERR_INVALID_PARAMETER
+              CIM_ERR_NOT_FOUND
+        """
+        iname = params['InstanceName']
+        if not iname.namespace:
+            iname.namespace = namespace
+
+        inst = self._get_instance(iname, namespace, **params)
+
+        return self._make_tuple([inst])
+
+    def _fake_deleteinstance(self, namespace, **params):
+        """
+        Implements a mock server reactor for
+        :meth:`~pywbem.WBEMConnection.DeleteInstance`.
+
+        This deletes a single instance from the mock repository based on the
+        iname and namespace parameters.
+        """
+        iname = params['InstanceName']
+        iname.namespace = namespace
+
+        try:
+            self.instances[namespace]
+        except KeyError:
+            ce = CIMError(CIM_ERR_NOT_FOUND,
+                          'Namespace %s not found' % namespace)
+            raise ce
+
+        del_inst = None
+        for enum_tup in enumerate(self.instances[namespace]):
+            index = enum_tup[0]
+            inst = enum_tup[1]
+            if iname == inst.path:
```

Comment:
That could be another use of the instance lookup by path function I proposed above.

Reply:
I still have cleanup on this.  Want to make  general function for each repo to set and delete.